addons/mail/static/src/new/thread/link_preview/link_preview_list.js

```diff
@@ -0,0 +1,50 @@
+/** @odoo-module */
+
+import { useMessaging } from "../../messaging_hook";
+import { Component } from "@odoo/owl";
+import { LinkPreviewCard } from "./link_preview_card";
+import { LinkPreviewImage } from "./link_preview_image";
+import { LinkPreviewVideo } from "./link_preview_video";
+
+export class LinkPreviewList extends Component {
+    setup() {
+        this.messaging = useMessaging();
+    }
+
+    // TODO Could somehow directy use the
+    get linkPreviews() {
+        return Object.values(this.messaging.linkPreviews).filter(
+            (linkPreview) => linkPreview.message_id === this.props.message.id
+        );
+    }
```

Comment:
Quite inefficient. You could store on `message` something like `linkPreviewIds` or something.